src/main/java/filter/Parser.java

```diff
@@ -126,6 +127,8 @@ private FilterExpression parseKeyword(Token token) {
 
     private int getInfixPrecedence(Token token) {
         switch (token.getType()) {
+        case SEMICOLON:
+            return Precedence.SHORTER_DISJUNCTION;
```

Comment:
Hmm, is this needed? This is for operators between qualifiers, which are taken care of by the main loop in `parseExpression`. `;` isn't handled there as it's only ever used in the context of a single qualifier.

The parse exception in the `sort:label;assignee` case is due to `;` not being a recognised infix operator, not because `sort` doesn't parse properly.